pyproject.toml

```
@@ -0,0 +1,126 @@
+classifiers = [
+    'Development Status :: 2 - Pre-Alpha',
+    'Intended Audience :: Developers',
+    'License :: Free for non-commercial use',
+    'Natural Language :: English',
+    'Programming Language :: Python :: 3',
+    'Programming Language :: Python :: 3.8',
+    'Programming Language :: Python :: 3.9',
+    'Programming Language :: Python :: 3.10',
+    'Programming Language :: Python :: 3.11',
+    'Topic :: Scientific/Engineering :: Artificial Intelligence',
+]
+keywords = ['ctgan', 'CTGAN']
+version = '0.9.1.dev0'
+license = { text = 'BSL-1.1' }
+requires-python = '>=3.8,<3.12'
+readme = 'README.md'
+dependencies = [
+    "numpy>=1.20.0;python_version<'3.10'",
+    "numpy>=1.23.3;python_version>='3.10'",
+    "pandas>=1.1.3;python_version<'3.10'",
+    "pandas>=1.3.4;python_version>='3.10' and python_version<'3.11'",
+    "pandas>=1.5.0;python_version>='3.11'",
+    "scikit-learn>=0.24;python_version<'3.10'",
+    "scikit-learn>=1.1.3;python_version>='3.10'",
+    "torch>=1.8.0;python_version<'3.10'",
+    "torch>=1.11.0;python_version>='3.10' and python_version<'3.11'",
+    "torch>=2.0.0;python_version>='3.11'",
+    'tqdm>=4.15',
+    'rdt>=1.6.1',
+]
+
+[project.urls]
+"Source Code"= "https://github.com/sdv-dev/CTGAN/"
+"Issue Tracker" = "https://github.com/sdv-dev/CTGAN/issues"
+"Changes" = "https://github.com/sdv-dev/CTGAN/blob/main/HISTORY.md"
+"Twitter" = "https://twitter.com/sdv_dev"
+"Chat" = "https://bit.ly/sdv-slack-invite"
+
+[project.entry-points]
+ctgan = { main = 'ctgan.cli.__main__:main' }
+
+[project.optional-dependencies]
+test = [
+    'pytest>=3.4.2',
+    'pytest-rerunfailures>=9.1.1,<10',
+    'pytest-cov>=2.6.0',
+    'rundoc>=0.4.3,<0.5',
+    'pytest-runner >= 2.11.1',
+    'tomli>=2.0.0,<3',
+]
+dev = [
+    'ctgan[test]',
+
+    # general
+    'pip>=9.0.1',
+    'bumpversion>=0.5.3,<0.6',
+    'watchdog>=0.8.3,<0.11',
+
+    # style check
+    'flake8>=3.7.7,<4',
+    'isort>=4.3.4,<5',
+    'dlint>=0.11.0,<0.12',  # code security addon for flake8
+    'flake8-debugger>=4.0.0,<4.1',
+    'flake8-mock>=0.3,<0.4',
+    'flake8-mutable>=1.2.0,<1.3',
+    'flake8-absolute-import>=1.0,<2',
+    'flake8-multiline-containers>=0.0.18,<0.1',
+    'flake8-print>=4.0.0,<4.1',
+    'flake8-quotes>=3.3.0,<4',
+    'flake8-fixme>=1.1.1,<1.2',
+    'flake8-expression-complexity>=0.0.9,<0.1',
+    'flake8-eradicate>=1.1.0,<1.2',
+    'flake8-builtins>=1.5.3,<1.6',
+    'flake8-variables-names>=0.0.4,<0.1',
+    'pandas-vet>=0.2.2,<0.3',
+    'flake8-comprehensions>=3.6.1,<3.7',
+    'dlint>=0.11.0,<0.12',
+    'flake8-docstrings>=1.5.0,<2',
+    'flake8-sfs>=0.0.3,<0.1',
+    'flake8-pytest-style>=1.5.0,<2',
+
+    # fix style issues
+    'autoflake>=1.1,<2',
+    'autopep8>=1.4.3,<1.6',
+
+    # distribute on PyPI
+    'twine>=1.10.0,<4',
+    'wheel>=0.30.0',
+
+    # Advanced testing
+    'coverage>=4.5.1,<6',
+    'tox>=2.9.1,<4',
+
+    'invoke',
+]
+
+[tool.setuptools]
+include-package-data = true
+
```

One thing I forgot, you'll need this:
```suggestion
[tool.setuptools.dynamic]
version = {attr = 'ctgan.__version__'}
```